### ISSUE TITLE
Weird behavior while changing the Batch size

### ISSUE THREAD

**Issue by eslambakr** [dbolya/yolact#325]
Hello @dbolya ,

Thanks for your amazing work.

I have a question about batch size, when I changed the backbone to EfficientNet instead of using the default one I got a strange results. with some debugging I noticed that the training is going well while the batch size is 6 or higher and when I set it to a small number because a hardware limitation The losses stack at certain high value and the mAP is zero however I let the training for 60 epoch and the mAP is still zero.

When I checked in the code what is the effect of the batchsize I didn't notice any imporatnt thing except you set `cfg.freeze_bn = True`  

So I rerun the expereiment which was with small batch size (=2) but this time I put it to False `cfg.freeze_bn = False` but unfortuantely, The loss becomes a bigger number and the mAP still zero
But the same experiment works fine when I increased the batch size from 2 to 6

So what do you think about that, what may cause this weird behavior?

Thanks in advance.

**Comment by dbolya** [comment]
So batch norm is automatically frozen when the batch size is < 6, as (I think) you're pointing out. When made your changes, did you do that here:
https://github.com/dbolya/yolact/blob/eee5ce36242fa1de808a287a3cfe6055486e42b5/train.py#L115-L118

If, so that's weird. Are you training from scratch or are you loading pretrained weights? Note that if your file doesn't load properly, I think any errors get suppressed. So if you're loading pretrained weights, make sure to confirm you're loading the right ones. And if you're not loading pretrained weights (or there's an issue with loading them), freezing batch norm would produce the effect you're experiencing.
